app/build.gradle

```diff
@@ -139,6 +137,12 @@ kotlin {
     experimental { coroutines 'enable' }
 }
 
+tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
```

Comment:
nit: I actually don't know that much gradle – why does this diverge from [the docs](https://kotlinlang.org/docs/reference/whatsnew12.html#warnings-as-errors), which state:
```groovy
compileKotlin {
    kotlinOptions.allWarningsAsErrors = true
}
```

Furthermore, I'd expect tasks.CompileKotlin, rather than `KotlinCompile` – why is it KotlinCompile?

Reply:
The documentation seems to be aimed at pure Java + Kotlin projects, and since Android's gradle implementation produces tasks with a different naming schema, it threw a "Could not find compileKotlin()..." error and I based my code on the solution proposed [here](https://stackoverflow.com/a/44143253). I tried tasks.CompileKotlin now and it results in an error("Could not get unknown property 'org' for project..."). I couldn't find a reason why, the closest I got was this [official doc](https://kotlinlang.org/docs/reference/using-gradle.html) while it doesn't have an explanation it uses the same code(under Compiler Options).

Reply:
This makes sense – we don't have the compileKotlin task due to the build variants and instead we have e.g. compileFocusWebviewDebugKotlin so we assign it to all tasks.

Thanks for the sources! It's great that I'm learning about gradle by reviewing this too! :)